packages/fuels-types/src/param_types.rs

```diff
@@ -68,6 +68,40 @@ impl ParamType {
         }
     }
 
+    pub fn contains_no_nested_vectors(&self) -> bool {
+        match &self {
+            ParamType::Vector(param_type) | ParamType::Array(param_type, ..) => {
+                !param_type.is_vector() && param_type.contains_no_nested_vectors()
+            }
+            ParamType::Struct {
+                fields, generics, ..
+            } => {
+                fields.iter().all(|(_, param_type)| {
+                    !param_type.is_vector() && param_type.contains_no_nested_vectors()
+                }) && generics.iter().all(|param_type| {
+                    !param_type.is_vector() && param_type.contains_no_nested_vectors()
+                })
+            }
+            ParamType::Tuple(param_types, ..) => param_types.iter().all(|param_type| {
+                !param_type.is_vector() && param_type.contains_no_nested_vectors()
+            }),
+            ParamType::Enum {
+                generics, variants, ..
+            } => {
+                generics.iter().all(|param_type| {
+                    !param_type.is_vector() && param_type.contains_no_nested_vectors()
+                }) && variants.param_types().iter().all(|param_type| {
+                    !param_type.is_vector() && param_type.contains_no_nested_vectors()
+                })
+            }
+            _ => true,
+        }
+    }
```

Comment:
This could be cleaned up a bit.

```rust
    pub fn contains_nested_vectors(&self) -> bool {
        match &self {
            ParamType::Vector(param_type) => param_type.uses_vectors(),
            _ => self.uses_vectors(),
        }
    }

    fn uses_vectors(&self) -> bool {
        match &self {
            ParamType::Vector(..) => true,
            ParamType::Array(param_type, ..) => param_type.uses_vectors(),
            ParamType::Tuple(param_types, ..) => Self::any_nested_vectors(param_types),
            ParamType::Struct {
                fields, generics, ..
            } => {
                let fields = fields.iter().map(|(_, param_type)| param_type);
                Self::any_nested_vectors(chain!(fields, generics))
            }
            ParamType::Enum {
                generics, variants, ..
            } => {
                let variants = variants.param_types();
                Self::any_nested_vectors(chain!(&variants, generics))
            }
            _ => false,
        }
    }

    fn any_nested_vectors<'a>(param_types: impl IntoIterator<Item = &'a ParamType>) -> bool {
        param_types
            .into_iter()
            .any(|param_type| param_type.uses_vectors())
    }
```

Reply:
This takes in consideration that you changed `contains_no_nested_vectors`

Reply:
Thanks for updating the code. However, The `is_vector()` check is needlessly repeated and can be avoided if you use a separate function as we suggested above. Also, chaining the iterators is cleaner than calling the `any_nested_vectors` several times.